CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ set( SEMIHOSTING ON )
 set( STACK_SIZE 2 ) # In KB
 set( RAM_SIZE 0x100000 ) # 1MB in bytes
 set( LINKER_SCRIPT "kernel.ld" )
+set( VIRT_DRIVER ON )
```

Comment:
I would rename this SERIAL_DRIVER since we've got more than one type now.

Then for your RPi stuff, you set SERIAL_DRIVER=src/hw/driver/<driver>.c